packages/events/src/service/events/events.ts

```diff
@@ -120,6 +120,85 @@ async function deleteEventAttachments(token: string, event: EventDocument) {
   }
 }
 
+async function cleanUnreferencedAttachmentsFromPreviousDrafts(
+  token: string,
+  event: EventDocument,
+  action: ActionDocument
+) {
+  if (action.draft) {
+    return
+  }
+  const previousDrafts = event.actions.filter(
+    (x: ActionDocument) =>
+      x.type === action.type && x.createdBy === action.createdBy && x.draft
+  )
```

Comment:
Only take actions from the end of the list until you encounter one where the condition doesn't apply. Otherwise it can happen that we remove attachments referred to in drafts made a long time ago and not just in the drafts before this action was "committed"